File: frontend/src/api/compliance.ts
/**
 * 合规：导出审批、审计日志、策略说明
 */
import http from './client'

export interface ExportRequestOut {
  id: number
  applicant_id: number
  case_id: number
  person_id: string
  file_format: string
  status: string
  reviewer_id: number | null
  review_note: string | null
  reviewed_at: string | null
  created_at: string
}

export interface AuditLogOut {
  id: number
  user_id: number | null
  case_id: number | null
  action: string
  resource_type: string
  resource_id: string | null
  ip_address: string | null
  detail: Record<string, unknown> | null
  created_at: string
}

export function createExportRequest(body: {
  case_id: number
  person_id: string
  file_format: 'pdf' | 'docx'
}) {
  return http.post<ExportRequestOut>('/compliance/export-requests', body)
}

export function listExportRequests(scope: 'mine' | 'all' = 'mine') {
  return http.get<ExportRequestOut[]>('/compliance/export-requests', {
    params: { scope },
  })
}

export function approveExportRequest(id: number, note?: string) {
  return http.post<ExportRequestOut>(`/compliance/export-requests/${id}/approve`, { note })
}

export function rejectExportRequest(id: number, note?: string) {
  return http.post<ExportRequestOut>(`/compliance/export-requests/${id}/reject`, { note })
}

export function listAuditLogs(params?: {
  user_id?: number
  case_id?: number
  action_prefix?: string
  limit?: number
}) {
  return http.get<AuditLogOut[]>('/compliance/audit-logs', { params })
}

export function getComplianceSettings() {
  return http.get<{ export_approval_required: boolean; masking: Record<string, string> }>(
    '/compliance/settings',
  )
}

File: backend/app/schemas/compliance.py
"""合规：导出审批、审计日志 API 契约。"""

from typing import Any, Literal

from pydantic import BaseModel, Field


class ExportRequestCreate(BaseModel):
    case_id: int = Field(ge=1)
    person_id: str = Field(min_length=1, max_length=256)
    file_format: Literal["pdf", "docx"] = "pdf"


class ExportRequestOut(BaseModel):
    id: int
    applicant_id: int
    case_id: int
    person_id: str
    file_format: str
    status: str
    reviewer_id: int | None = None
    review_note: str | None = None
    reviewed_at: str | None = None
    created_at: str


class ExportReviewBody(BaseModel):
    note: str | None = None


class AuditLogOut(BaseModel):
    id: int
    user_id: int | None
    case_id: int | None
    action: str
    resource_type: str
    resource_id: str | None
    ip_address: str | None
    detail: dict[str, Any] | None
    created_at: str
